### PR TITLE
Updated serializers and views to use django-rest-framework >= 3.0

### PR DESCRIPTION
**Pull request by maerteijn**

### CODE REVIEW

**Comment by maerteijn** [comment]
Will propably need some review as a lot of things changed in django-rest-api v3:
http://www.django-rest-framework.org/topics/3.0-announcement/

**Comment by jmound** [comment]
I got this error from a new setup, a catalog of just one item. I wouldn't be surprised if it's a mistake on the setup on my side. 

```
Request Method: GET
Request URL: http://localhost:8000/oscarapi/products/

Traceback:
File ".../venvs/blnrd/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  111.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File ".../venvs/blnrd/lib/python2.7/site-packages/django/views/decorators/csrf.py" in wrapped_view
  57.         return view_func(*args, **kwargs)
File ".../venvs/blnrd/lib/python2.7/site-packages/django/views/generic/base.py" in view
  69.             return self.dispatch(request, *args, **kwargs)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  452.             response = self.handle_exception(exc)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/views.py" in dispatch
  449.             response = handler(request, *args, **kwargs)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/generics.py" in get
  199.         return self.list(request, *args, **kwargs)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/mixins.py" in list
  41.         page = self.paginate_queryset(queryset)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/generics.py" in paginate_queryset
  170.         return self.paginator.paginate_queryset(queryset, self.request, view=self)
File ".../venvs/blnrd/lib/python2.7/site-packages/rest_framework/pagination.py" in paginate_queryset
  299.             self.page = paginator.page(page_number)
File ".../venvs/blnrd/lib/python2.7/site-packages/django/core/paginator.py" in page
  55.         return self._get_page(self.object_list[bottom:top], number, self)

Exception Type: TypeError at /oscarapi/products/
Exception Value: 'ProductManager' object has no attribute '__getitem__'
```

**Comment by specialunderwear** [comment]
I've got the feeling that it caused by queryset being specified as Product.objects in the ProductSerializer. There is nothing wrong with doing that but maybe they broke something in rfw3.

File: oscarapi/views/basic.py [hunk 1]
```diff
@@ -98,44 +108,57 @@ def get(self, request, pk=None, format=None):
         return Response(ser.data)
 
 
-class ProductAvailability(generics.RetrieveAPIView):
-    model = Product
-    serializer_class = serializers.ProductAvailabilitySerializer
+class ProductAvailability(APIView):
```
Comment (specialunderwear) [inline oscarapi/views/basic.py]: Why is all this needed?

Is RetriveAPIView no longer supported or something?

Comment (maerteijn) [inline oscarapi/views/basic.py]: Yeah, but my implementation of the serializer class was ugly so now at least this is just one way to do it. Could try to rewrite both get methods though.

**Comment by maerteijn** [comment]
@jmound: could you specify what version of django you used and what kind of item you added to the catalog?

**Comment by maerteijn** [comment]
I tested the sandbox with Django 1.6.10 and 1.7.6 and I can't reproduce the problem:
```
git clone -b restframework3 git@github.com:django-oscar/django-oscar-api.git
cd django-oscar-api
make sandbox
python sandbox/manage.py createsuperuser
python sandbox/manage.py runserver
```
Added a new product with the admin and viewed it with /api/products/

**Comment by specialunderwear** [comment]
Aha, sound like a good point, I can not merge this this week btw.
I don't have time to test until next week.

**Comment by jmound** [comment]
@maerteijn 
```
$ pip freeze
Babel==1.3
Django==1.7.6
django-appconf==1.0.1
django-compressor==1.4
django-extensions==1.5.1
django-extra-views==0.6.4
django-haystack==2.3.1
django-oscar==1.0.2
-e git+git@github.com:django-oscar/django-oscar-api.git@d539f639657de9644aff098a137a96e764823c02#egg=django_oscar_api-origin_restframework3
django-tables2==0.15.0
django-treebeard==2.0
djangorestframework==3.1.0
factory-boy==2.4.1
gunicorn==19.3.0
mock==1.0.1
phonenumbers==6.3.0
Pillow==2.4.0
psycopg2==2.6
purl==1.0.3
pytz==2014.10
six==1.9.0
sorl-thumbnail==11.12.1b0
Unidecode==0.4.17
Werkzeug==0.10.1
```
python 2.7.6
I ran through your steps in a fresh virtualenv and it worked fine. I haven't been able to find the difference between my site and the sandbox yet, but it's likely there.

**Comment by maerteijn** [comment]
@jmound Ok, let me know if you have a clue or know what is, I'm curious now :)

**Comment by jmound** [comment]
Finally found it. I narrowed the problem to something in my settings file. After some tedious debugging:
```
REST_FRAMEWORK = {
    'PAGE_SIZE': 10,
    ...
```
reliably causes the problem. I had originally grabbed it from the DRF [quickstart](http://www.django-rest-framework.org/tutorial/quickstart/) page.

**Comment by specialunderwear** [comment]
I think you found a bug in rest framework :)

**Comment by michaelkuty** [comment]
How looks this merge ? Please can be merged ? 
I'm eager for this merge
+1

**Comment by specialunderwear** [comment]
If this gets rebased on the checkout branch, you got what you need.
It will still take me a while to merge this.

@maerteijn can you rebase this on the checkout branch?

**Comment by jakobholmelund** [comment]
Will this get merged soon ?

**Comment by michaelkuty** [comment]
please we need DRF 3 +

**Comment by jakobholmelund** [comment]
Just merged this myself, and seems to be working. At least all links from the api overview page works.. So if you like me, just can't wait for this, then checkout master from here https://github.com/sonofatailor/django-oscar-api . I've made a change in the /serializers/login.py because i use email instead of username for logins. So change this yourself.

**Comment by specialunderwear** [comment]
Yeah I'm sorry, but still no time to merge this.
Most likely after june 10 there will be some time.
Also this has to be rebased on master @maerteijn

**Comment by michaelkuty** [comment]
+1

**Comment by specialunderwear** [comment]
Doe eens dan @maerteijn

**Comment by specialunderwear** [comment]
I updated this pull request by merging it into master and then adding fixes for everything that was broken.

**Comment by specialunderwear** [comment]
If all goes well, it will be merged next monday!!!!

**Comment by michaelkuty** [comment]
awesome news

**Comment by specialunderwear** [comment]
IT WAS MERGED!!!!!!!!!!!!!!!!!!!

**Pull request merged**